docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+# The autorenew feature
+
+## Goal
+-	Allow Hedera Services to automatically search for entities such as accounts, files, smart contracts, topics, tokens... that are about to expire and renew them.
+
+## Design
+- Introduce new settings in `application.properties`:
+  * `autorenew.isEnabled`
+  * `autorenew.numberOfEntitiesToCheck`
+  * `autorenew.maxNumberOfEntitiesToRenew`
+  * `autorenew.gracePeriod`
```

Comment:
When an account is in the gracePeriod does the entity have a property that highlights this fact?
Some kind of entity state might be important to expose this when info is pulled on it.
Eventually this might be something that makes its way to exchanges/users and allows them to alert users or take action themselves.

Reply:
I think this is a good enhancement. We have to change the protobuf to add this field though.

Reply:
Is the alternative that a client would need to verify the `expirationTime` on the entity to know if it is in gracePeriod?